### DISCUSSION TITLE
Document icons issues + adaptation to new Apple's HIG

### DISCUSSION THREAD

**Discussion by Zabriskije** [iina/iina#3670] [Ideas]
Hi folks,
the document icons have some aesthetic issues and don't correspond to the current HIG (I resized the images to make the post more readable; if you need it, you can click them, or right-click + open in a new panel to view them in their original size):

<img width="150" src="https://user-images.githubusercontent.com/101254295/159226376-a238be88-ec68-40ea-997e-edd9f969524e.png">  <img width="150" src="https://user-images.githubusercontent.com/101254295/159226385-5bec74d8-9d4b-4ee0-b8b1-889feaf413c6.png">  <img width="150" src="https://user-images.githubusercontent.com/101254295/159226390-a50977e8-98a6-463c-9c63-da38995777de.png">  <img width="150" src="https://user-images.githubusercontent.com/101254295/159226391-85f03bbd-db86-4e8c-a4c0-cc6e72d444d7.png">  <img width="150" src="https://user-images.githubusercontent.com/101254295/159226393-c6a8c738-fdc9-432c-bb0c-8ee8eeff613d.png">

1. The page's folded corner is now being replaced with a flatter one;
2. There are some aesthetic marks left over from previous unpolished layers (I think?);
3. Same as above;
4. My theory is that there are two separate layers that have not been masked correctly (or other common methods), and that results in visual errors like the grey line on the left and bottom side of the purple-blu layer, and a general sense of misalignment;
5. The area of the possible rectangle under the purple-blue layer causing the upper issue seen in image 2.

I've made new assets to correct those issues, if you like to incorporate them in future updates let me know, so I can understand exactly what you need (I don't use Xcode, but I've read that you can create the icons directly from there when building).

The page layer, logo, and font have been adapted to the current HIG.

<img width="256" src="https://user-images.githubusercontent.com/101254295/159230034-2aa59d1a-55d6-40f9-8056-aae13d7daa0b.png"> <img width="256" src="https://user-images.githubusercontent.com/101254295/159230038-8bb6dfb9-d620-4990-b2b8-505dec25bbdb.png"> <img width="256" src="https://user-images.githubusercontent.com/101254295/159230042-24fcb13f-bebd-4045-ab4b-0a3e3443954b.png">

**Comment by low-batt** [comment]
Hello from another IINA user. 

I assume it is this section of the HIG that is applicable?: [Document Icons](https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/document-icons/)

IINA's [CONTRIBUTING](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md) document says:
> Stay consistent with the [macOS Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/macos/overview/themes/).

As should be expected IINA needs to adhere to the HIG and deviations from it need to be fixed at some point. In my opinion the document icons need to be updated.

An example of the actual icons being used can be found in the source code here: [doc_flv.iconset](https://github.com/iina/iina/tree/develop/iina/Assets.xcassets/DocIcons/doc_flv.iconset)

From that section of the HIG:
>If you don’t supply a document icon for a file type you support, macOS 11 creates one for you by compositing your app icon and the file’s extension onto the canvas.

And:
>If your app runs in earlier versions of macOS, you should supply a legacy document icon. If you let earlier versions of macOS generate a document icon for you, you get different results in different versions. Specifically, macOS 10.15 creates a document icon by compositing your app icon, but not the file extension; macOS 10.14 and earlier use a generic document background if you don’t supply any assets.

Could IINA be changed to just rely upon the system generated icons? Currently IINA supports macOS 10.11+. Maybe versions of macOS older than 10.14 require you to supply the icons? If that is the case could IINA be structured to use the system generated icons for macOS 11+ and use the existing icons for older versions of macOS? That would be the kind of questions I'd be looking for answers to.

This is not the only icon related issue outstanding. Issue #3116 is about icons in toggle menus not obeying the HIG. Discussion #3631 is proposing a new logo/icon for the application. That might affect the look of the document icons.

At the moment the IINA developers need to be focused on getting out a bug fix release. But in my opinion you have raised a valid issue that needs to be addressed at some point.

**Comment by Zabriskije** [comment]
> I assume it is this section of the HIG that is applicable?: [Document Icons](https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/document-icons/)

Yup

> Could IINA be changed to just rely upon the system-generated icons? Currently, IINA supports macOS 10.11+. Maybe versions of macOS older than 10.14 require you to supply the icons? If that is the case could IINA be structured to use the system-generated icons for macOS 11+ and use the existing icons for older versions of macOS?
> 
> Discussion https://github.com/iina/iina/discussions/3631 is proposing a new logo/icon for the application. That might affect the look of the document icons.
> 
> At the moment the IINA developers need to be focused on getting out a bug fix release. But in my opinion, you have raised a valid issue that needs to be addressed at some point.

Fair questions. From my understanding, previous icons were custom-made instead of relying on Xcode automatic generation. But either way, I'm happy to provide custom-made icons as well as graphic assets to make dev's life simpler and resolve those issues.

I'm also willing to generate all graphic assets to match the current logo style, so in the next update the HIG are respected.
After that, we can discuss new logos and I'll have no problem reworking my files and updating the assets.

**Comment by Zabriskije** [comment]
While I'm here, I could also provide a cleaner version of the current app icon, that presents some old/wrong visual elements:

<img width="150" src="https://user-images.githubusercontent.com/101254295/159490602-10cc1f02-3201-4c15-8128-ee27173665f3.png"> <img width="150" src="https://user-images.githubusercontent.com/101254295/159490619-4bf71851-71d5-4836-9655-ea988b298213.png"> <img width="150" src="https://user-images.githubusercontent.com/101254295/159490627-b2798cc2-d71e-4073-988d-0feca23ce2e9.png">

1. The white inner-top line is now disused and refers to old MacOS/iOS visual styles;
2. Such a detailed background pattern is not rendered at small view sizes;
3. Icon shadow is incorrectly placed at the perfect center of the above layer, resulting in uniform 4-side shadows; whereas instead that should be offset towards the bottom, resulting in more shadow coming from below the canvas. This also affects the visual perception of the icon in the dock and launcher, making it feel smaller or different from other icons.

Here's the polished one:

<img width="256" src="https://user-images.githubusercontent.com/101254295/159492774-8772700f-76df-435d-a9d7-d24b249d4726.png">

Dark background colors are also a big part of the icon feeling smaller, and should be considered to make them brighter.
Like that, for example:

<img width="256" src="https://user-images.githubusercontent.com/101254295/159495062-01be1f74-f9f9-4130-b1d4-af59ec996202.png">

**Comment by low-batt** [comment]
Sorry I'm slow to respond. Caught me busy this week. Should have more time this weekend.

The application icon is a separate issue. I've looked at the [App Icon](https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/app-icon/) section of the HIG and just as you indicated I do not see the attributes you pointed out in the IINA icon being used in the latest HIG examples. Are you using the HIG [Apple Design Resources](https://developer.apple.com/design/resources/#macos-apps) to produce your examples?

Seems to me like there are two separate issues, the document icons and the application icon. Both represent deviations from the HIG, so in my opinion both need to be addressed.

My impression on the situation with the document icons is that macOS (not Xcode) will create them for your application if you don't supply them. I'll run some tests this weekend and see how macOS 12 behaves. Possibly we won't need new document icons.

Sure looks like the application icon needs to be updated to follow the latest HIG. If given the icon PNG files I can deal with preparing a pull request with updated icons. Lets first see what the developers think about this. I'll see if I can get the developers attention and get their take on this.

**Comment by Zabriskije** [comment]
No problem @low-batt, I appreciate you're taking the time to respond and understanding the issue.

> I've looked at the [App Icon](https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/app-icon/) section of the HIG and just as you indicated I do not see the attributes you pointed out in the IINA icon being used in the latest HIG examples. Are you using the HIG [Apple Design Resources](https://developer.apple.com/design/resources/#macos-apps) to produce your examples?

Yup, a Pixelmator .pxd format 1:1 I've made from the Photoshop .psd provided for the App Icon, and my best-attempt-replica for the document icon, because in Apple Design Resources there's no background-free version of it, its intended use is to have a reference for creating and exporting assets to use in Xcode, not to be an actual asset (it have no transparent background and there are no provided layers to work under the fold so you will end up covering it and the relative shadow).

> My impression of the situation with the document icons is that macOS (not Xcode) will create them for your application if you don't supply them. I'll run some tests this weekend and see how macOS 12 behaves. Possibly we won't need new document icons.

In this case, the icons that will be used are the default from MacOS, and I think the developers want to have them customized ― and I kinda like that, although sometimes the system like to display the default document icon for a specific extension and not the one provided (as sometimes happens for .html and others file-type in VSCode, Atom, etc.).

Let me know about the test, and we'll see what the devs say.

<br>
<hr>
<br>

Just to be clear for other readers (from Apple's [App Icon](https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/app-icon/) section of HIG):

> To download templates that specify the correct shape and drop shadow, see [Apple Design Resources](https://developer.apple.com/design/resources/#macos-apps)

> Use the drop shadow in the icon-design [template](https://developer.apple.com/design/resources/#macos-apps). The template includes the system-defined drop shadow that helps your app icon coordinate with other macOS 11 icons.

For the upper line and the background, there isn't exactly a written rule, just a general ― kind of ― common practice to move towards cleaner looking visuals:

> Embrace simplicity. Find a concept or element that captures the essence of your app and express it in a simple, unique way, adding details only when doing so enhances meaning. Too many details can be hard to discern and can make the icon appear muddy, especially at smaller sizes.

> Consider using interior shadows and highlights to add definition and realism.

A solid or bold light-line is normally used when highlighting features of a realistic-looking object or when the shape of the icon itself is being used as a real object:

<img width="128" src="https://user-images.githubusercontent.com/101254295/159890151-21d16c79-7c53-4da4-99d1-9f90fe215f72.png">

If you look at other Apple icons, it is not used in any other case.

**Comment by low-batt** [comment]
I had a brief chat with one of the IINA developers about the icon issues. As I suspected this not a good time to bring this up. The developers are currently way behind and totally overloaded dealing with reviewing and merging a huge backlog of changes for issues such as crashes. Trying to get them interested in updating icons to meet new HIG requirements is a tough sell at this moment. It is like trying to talk to a mechanic about the tint of the car being slighly off when they suspect the engine has seized, the transmission has failed and an axle or two is broken.

If you want to create some updated corrected icons I will hang on to them and try and get them into IINA once the project has caught up and has time to consider cosmetic fixes. I definitely agree IINA needs to be updated to meet the current HIG requirements for icons.

I did run an experiment to confirm macOS will create a document icon if an app does not supply one. I deleted the `CFBundleTypeIconFile` entry for the FLV file type from Info.plist. Before I did that this is what the document icon looked like:
![Screen Shot 2022-03-26 at 11 40 51 PM](https://user-images.githubusercontent.com/86170219/160320840-9896fd4c-3ef1-4894-881b-9fac5c84f261.jpg)

After making that change macOS 12 supplied this icon:
![Screen Shot 2022-03-26 at 11 44 05 PM](https://user-images.githubusercontent.com/86170219/160320872-28f6389c-49af-46df-9e2a-5bc79db08630.jpg)

Although macOS will generate document icons for an app, the HIG indicates you should not take advantage of that feature and should include document icons if you application supports older versions of macOS.  At least that is how I interpret this statement from the [Document Icons](https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/document-icons/) section of the HIG:
> If your app runs in earlier versions of macOS, you should supply a legacy document icon. If you let earlier versions of macOS generate a document icon for you, you get different results in different versions. Specifically, macOS 10.15 creates a document icon by compositing your app icon, but not the file extension; macOS 10.14 and earlier use a generic document background if you don’t supply any assets.

IINA currently supports macOS 10.11+. So I don't think IINA can rely upon macOS to generate document icons. New document icons are needed as well as a new application icon.

**Comment by Zabriskije** [comment]
> I had a brief chat with one of the IINA developers about the icon issues. As I suspected this not a good time to bring this up. The developers are currently way behind and totally overloaded dealing with reviewing and merging a huge backlog of changes for issues such as crashes. Trying to get them interested in updating icons to meet new HIG requirements is a tough sell at this moment. It is like trying to talk to a mechanic about the tint of the car being slighly off when they suspect the engine has seized, the transmission has failed and an axle or two is broken.

Totally understandable.

> If you want to create some updated corrected icons I will hang on to them and try and get them into IINA once the project has caught up and has time to consider cosmetic fixes. I definitely agree IINA needs to be updated to meet the current HIG requirements for icons.

I'll post them here once done. ETA ~2 weeks.

**Comment by Zabriskije** [comment]
>~~I'll post them here once done. ETA 2 weeks.~~

Done, I had some spare time today: [IINA.zip](https://github.com/iina/iina/files/8362906/IINA.zip)

The archive contains all the current document icons, named as such, same colors; plus the .png and .icns of the polished logo (both dark and bluish as seen [here](https://github.com/iina/iina/discussions/3670#discussioncomment-2415014).
The logo used in the document icons is the outline of the [original IINA arrow](https://github.com/iina/iina/blob/ab7a3bda0d7d0f9f3b9661266480e20fb1676fde/iina/Assets.xcassets/Icons/iina_arrow.imageset/iina-arrow.png); while in the app icon is used in its original form.
For readability purposes: icon_16x16, icon_16x16@2, and icon_32x32 have been changed so they don't have text in them.

**Comment by low-batt** [comment]
Thanks! I've grabbed the zip file. I'll look into it tomorrow.

**Comment by low-batt** [comment]
I'm stuck on how to extract all the different size icons as png files from the .icns files. All the examples of using sips and iconutil I have found on the net are showing how to create a .icns file from a set of .png files. Nothing about how to extract all the different sizes from the .icns file. No luck with Preview either. No matter which one I select Preview extracts the largest icon. To change IINA I need to update the files seen here: [AppIcon.appiconset](https://github.com/iina/iina/tree/develop/iina/Assets.xcassets/AppIcon.appiconset)

**Comment by Zabriskije** [comment]
In preview, you can extract single assets from an icns by showing miniatures (⌥ ⌘ 2) and dragging them, but they're in tiff format.
I've made the iconset from the png in the zip file shared above: [icon.iconset.zip](https://github.com/iina/iina/files/8389321/icon.iconset.zip)

**Comment by low-batt** [comment]
Had more time today and figured out how to extract an iconset from a icns file using terminal:
```bash
iconutil -c iconset IINA.icns
```

I was able to create my own build of IINA with the new icons. I have a commit ready in a branch of my fork: [low-batt branch d-3670](https://github.com/low-batt/iina/tree/d-3670)

Once the IINA developers have gotten to the point where they are ready to deal with cosmetic issues I will bring up the issue of the icons needing to be updated. That commit used the "regular" app icon, not the colored one. I won't forget to ask the developers if they want to use that version.

I did commit the updated icons to [IINA+](https://github.com/iina-plus/iina). That is a special **unofficial** fork for experimenting with the latest features and bug fixes. The latest build contains the updated icons.

If you have time please create a set of document icons for GIF files. IINA can play animated gif files, but does not appear in the list of applications when you right click in Finder and select `Open With`. I plan on proposing a fix for that. Would be nice to include document icons for gif files in that fix.

**Comment by Zabriskije** [comment]
> I was able to create my own build of IINA with the new icons. I have a commit ready in a branch of my fork: [low-batt branch d-3670](https://github.com/low-batt/iina/tree/d-3670)
>
> Once the IINA developers have gotten to the point where they are ready to deal with cosmetic issues I will bring up the issue of the icons needing to be updated. That commit used the "regular" app icon, not the colored one. I won't forget to ask the developers if they want to use that version.

Awesome. Good job.

> I did commit the updated icons to [IINA+](https://github.com/iina-plus/iina). That is a special unofficial fork for experimenting with the latest features and bug fixes. The latest build contains the updated icons.

I didn't know about that, I'll try it later.

> If you have time please create a set of document icons for GIF files. IINA can play animated gif files, but does not appear in the list of applications when you right click in Finder and select Open With. I plan on proposing a fix for that. Would be nice to include document icons for gif files in that fix.

There you go: [doc_gif.zip](https://github.com/iina/iina/files/8396631/doc_gif.zip)
Also need .webp or other formats? Feel free to make a list of all the formats needed.

<img width="128" src="https://user-images.githubusercontent.com/101254295/161240484-507b06be-ad82-4998-b66a-76d0db3d8202.png">

**Comment by setanarut** [comment]
I think this style looks prettier than the custom diagonal ones. random vibrant colors mean nothing.

![160320872-28f6389c-49af-46df-9e2a-5bc79db08630](https://user-images.githubusercontent.com/36481442/161350953-4967c432-9b4e-4892-b674-dee10e76e0eb.jpg)

**Comment by low-batt** [comment]
Hi @hazarek!
Glad to hear your input. As the IINA developers are too busy with other issues to think about cosmetic ICON issues we are going with proposing updates that hopefully won't be rejected by the developers as they only "tweak" the existing IINA ICONs to match up with the HIG. So the new document ICONs preserve the random vibrant colors of the existing icons.

When the IINA developers get caught up we can see if they are amenable to more significant changes such as the app ICON that is not as dark that Zabriskije has proposed, and changing the document ICONs to not use those vibrant colors. There is also the proposal for a significant application ICON change in discussion #3631.

**Reply by setanarut** [reply]
Ok, I just said my cosmetic opinion. The rest is up to you. Thanks

**Reply by low-batt** [reply]
All good hazarek. I do appreciate you stating your opinion. Nice to know users care about such details. Ultimately IINA may end up going with the style you cited. Will have to see what the IINA developers think. But that has to wait for later. They are nose the grindstone working towards a new release and issues like crashes need to be addressed before polishing the look of IINA. So for the moment we must muddle forward as best we can without their guidance.

**Comment by low-batt** [comment]
The GIF ICONs have been used in PR #3680 to fix issue #3679. Thanks much for creating those ICONs! The same changes have been commited and pushed into  [IINA+](https://github.com/iina-plus/iina), so the latest build of that fork will show the new GIF document icons. Of course to see the ICON you need to specify IINA as the default application for GIF files and turn off "Show icon preview" in Finder, or use some other application for viewing the files.

On the offer to make additional ICONs, I'm thinking that without guidence from the IINA developers this is as far as we should take things at the moment.

**Comment by low-batt** [comment]
@Zabriskije The updated app icon and document icons you provided have been merged into the `develop` branch. They should be in the unsigned [nightly](https://nightly.iina.io/) builds. Unless I messed up.

I've been asked to check in with you and see if you have time to create two more app icons and a couple of buttons. Assuming you are interested and have time…

This has to do with issue #4383. We want to make it easier for users to tell they are running something other than the official release build. One convention for doing that is to use slightly different app icons for the different kinds of builds. See issue #4383 for examples. To do this we need:
- app icon for Beta build
- app icon for Nightly build
- button for Nightly build
- button for Debug build

Yes, the list seems odd. That is because we already have a special orange BETA button. #4383 has a picture of that. So we need a Nightly and a Debug button. We don't need a Debug app icon because we are going to use the normal app icon for that kind of build.

I asked about what guidelines I should give you and was told first ask for some simple mockups. 

My own thoughts are as follows…

Beta builds go out to normal users. So that icon has to be very explicit and obvious. I'm thinking it should have the word BETA on it. IINA already has an orange BETA button, so orange probably should be used. I think the Nightly build app icon can be more subtle.

Let me know if you are interested and have time to come up with a mockup.

**Comment by Zabriskije** [comment]
Sure, glad to help. Here they are:

<img width="128" src="https://user-images.githubusercontent.com/101254295/235142862-3c141f60-1f20-4cfa-bb73-8f6410f8f1ac.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235142872-0e3b603b-3db5-4611-8e37-d8a40d92fde9.png">

<img width="128" src="https://user-images.githubusercontent.com/101254295/235142933-d1efe098-487c-4d86-b39a-bbbee92ccd23.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235142943-04bb92ad-33c7-4d87-ae3d-74a792a91a49.png">

<img width="128" src="https://user-images.githubusercontent.com/101254295/235142898-6e8e730e-f6c8-478e-af76-72af3dac435a.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235142912-e68543ae-6b4d-4411-918f-5e708d6ac86f.png">

The third row uses different logo colors; could this be acceptable as a more elegant solution?
Otherwise, to follow your suggestions: row 1 image 1 could be used as beta, row 2 image 2 or row 3 image 2 as nightly.

About the buttons, if you're talking about the one visible in the #4384 image, here they are (I've also redone the beta one for consistency): <img width="36" src="https://user-images.githubusercontent.com/101254295/235146176-ab04664d-785a-4814-883c-4565db338ab0.svg"> <img width="57" src="https://user-images.githubusercontent.com/101254295/235146179-1e421e25-0314-4eaf-8017-949d94daf0b2.svg"> <img width="48" src="https://user-images.githubusercontent.com/101254295/235146177-11f6b1e5-7af0-43a4-8913-b6b42d0b258f.svg">

They're in SVG format, so if you right-click on them, you can download and use them right away.
Same goes for the icons, although they're just PNG @ 1024x1024.

**Reply by uiryuu** [reply]
Thank you very much.

I think the first line met my exception best, both the second and the third row don't tell the user explicitly what build they are using. But when the icon is small, we cannot really see the text clearly, so we should use the third row.

Just tested in my dock:
![image](https://user-images.githubusercontent.com/20237141/235270711-7c90d1d2-120c-4337-a90d-ab449bc30002.png)

I think the text and banner is too small when we zoom out a bit. I understand that enlarge the banner will overlap with the icon, so maybe we could consider slant banner like chrome:
![image](https://user-images.githubusercontent.com/20237141/235270968-ba381303-dd7a-4dd1-83be-d35493cef8fb.png)

As for the "button", they are not really images, they they are just text fields with colored background. So if you can share the exact color you are using that would be great.

**Reply by Zabriskije** [reply]
I'm not a big fan of slant banners, but here it is:

<img width="128" src="https://user-images.githubusercontent.com/101254295/235279055-76f2f2f5-26e9-4dfa-b64f-405e2bfcf7ed.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235279059-3b9b9316-598e-4d67-8713-0e29e10805ac.png">

What if we make the banner bigger™ and put it at the bottom instead?

<img width="128" src="https://user-images.githubusercontent.com/101254295/235278727-f8aa5c95-3f99-4b35-a944-8083c4e072e6.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235278728-c095cbbf-6a14-4370-977a-30a7f96f5d0c.png">

Hex colors (sRGB):
- Beta #FF8928
- Nightly #009FFE
- Debug #1BE1B4

**Reply by low-batt** [reply]
I like the looks of the bigger banner at the bottom.

**Reply by uiryuu** [reply]
![image](https://user-images.githubusercontent.com/20237141/235279850-92330cdf-8853-469d-8607-bef6b95bb866.png)

Let go with the larger banner one.

@Zabriskije Do you know how to make iconset? When the icon is small, we can fallback to these two:
 
<img width="264" alt="image" src="https://user-images.githubusercontent.com/20237141/235280379-1963061f-653b-45c5-a6f2-1a5efb787156.png">

**Reply by Zabriskije** [reply]
Yes, but I've used the right color shown in your last image for debug because it was too bright for the banner. If I use #009FFE — the one used in my last post — for smaller sizes, the logo is going to be the same as the original one (since I picked directly from it).

Just to make it simpler to understand:

Curren color #009FFE:

<img width="64" src="https://user-images.githubusercontent.com/101254295/235296802-f72fb532-5197-4c1e-9a6c-8405b6262bc0.png"> <img width="32" src="https://user-images.githubusercontent.com/101254295/235296743-05c6e990-52d6-49d7-9870-1d6eb0f17750.png"> <img width="16" src="https://user-images.githubusercontent.com/101254295/235296742-35b91281-0a8b-4856-a424-915783b33ef2.png">

Previously used #1BE1B4:

<img width="64" src="https://user-images.githubusercontent.com/101254295/235296796-200cc543-52c0-4f6b-b5b5-125ac7d79a38.png"> <img width="32" src="https://user-images.githubusercontent.com/101254295/235296795-a3c74f22-b8f5-4708-8c7c-2d3f9411a59f.png"> <img width="16" src="https://user-images.githubusercontent.com/101254295/235296793-7a84d743-be9b-49e7-8f68-fde62c0a3e82.png">

We can also use another color, such as #954DFF:

<img width="64" src="https://user-images.githubusercontent.com/101254295/235297793-f9284a50-e4b3-480a-a737-4cd0c8f04566.png"> <img width="32" src="https://user-images.githubusercontent.com/101254295/235297791-97b05d4e-c026-4b42-a39c-ff1adcbec9e8.png"> <img width="16" src="https://user-images.githubusercontent.com/101254295/235298155-9685dd4d-dbfe-4698-8bb0-33efd2b9f2b8.png">

This way we can differentiate it even at smaller sizes.
What's your take?

**Reply by uiryuu** [reply]
I like the second one, let's go with that.

**Reply by saagarjha** [reply]
If we're voting I would pick the colors since the banners are a bit small. This is what Safari does (blue/purple/black/gold for Release/Preview/Local/Profiling). It's also adopted by Chrome Canary.

**Reply by Zabriskije** [reply]
> If we're voting I would pick the colors since the banners are a bit small. This is what Safari does (blue/purple/black/gold for Release/Preview/Local/Profiling). It's also adopted by Chrome Canary.

+1, I'd also prefer using colors instead of banners.
Where can I see the Local/Profiling icons? I didn't find anything with a quick Google search.

Based on the colors listed, the result would be something like this (release, beta, nightly, debug):

<img width="128" src="https://user-images.githubusercontent.com/101254295/235351131-9f24946a-e5e8-4c0b-9d2d-6c4ba039da61.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235351125-ec1be368-28e2-4ce7-935b-af3c4f0a89b6.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235351129-dc6464f0-9191-44e2-ab2f-2e3c69f85c0f.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235351127-a14ffe6b-a844-4039-b667-926ba4d1dbd0.png">

The last two seem a bit odd to me, so I would still go for custom ones:

<img width="128" src="https://user-images.githubusercontent.com/101254295/235351131-9f24946a-e5e8-4c0b-9d2d-6c4ba039da61.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235351125-ec1be368-28e2-4ce7-935b-af3c4f0a89b6.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235351566-38b51988-fccd-4507-9c34-15badf0c6500.png"> <img width="128" src="https://user-images.githubusercontent.com/101254295/235351514-0f432c8f-572e-4ae8-8160-bf2e64f9195a.png">

**Reply by saagarjha** [reply]
I think they're internal. No need to match their colors, we can just pick whatever set we think looks good.

**Reply by low-batt** [reply]
Maybe only include the banner in the largest version of the icon?
Should be readable when displayed in Launchpad and on the About window?
This would re-enforce for the user what the color represents.
I agree once the icon gets small the banner is unreadable and should not be used.

**Reply by Zabriskije** [reply]
[Here's the `.iconset` with banners](https://github.com/iina/iina/files/11375629/IINA-builds.zip), added a debug one just in case

Colors: beta #FF8928 — nightly #954DFF — debug #1FD2AA

**Reply by low-batt** [reply]
The icon with the bigger banner is nice and readable in Launchpad:

![issue-4383-launch](https://user-images.githubusercontent.com/86170219/235725228-8c5890c9-25a0-4bb3-8c8a-7c708ef63c39.jpg)

Readable but small on the about window:

![issue-4383-about](https://user-images.githubusercontent.com/86170219/235725293-d4197152-167f-4dfb-8f84-316b2a597a37.jpg)

I defer to others as to whether or not to include a banner in the large icons.

**Reply by Zabriskije** [reply]
@low-batt I've updated the icons to lower a bit the logo since it feels out of center. You can find the new .iconset on my previous message.

KeyboardCleanTool 👀 so I'm not the only one using it

**Reply by low-batt** [reply]
Thanks!

@saagarjha and @uiryuu are the deciders. Need to hear from them and go with what they want.

Yes, `KeyboardCleanTool` is nice when trying to brush dust out of the keyboard. And I have `BetterTouchTool` configured so that I can control IINA using an Apple TV remote.

**Reply by uiryuu** [reply]
> [Here's the `.iconset` with banners](https://github.com/iina/iina/files/11375629/IINA-builds.zip), added a debug one just in case
> 
> Colors: beta #FF8928 — nightly #954DFF — debug #1FD2AA

LGTM, I think we can continue with these.

**Comment by low-batt** [comment]
Wow. That was fast. THANKS!

We are now waiting to see what @uiryuu thinks of the mockups.

**Comment by anohren** [comment]
The first two styles convey certain qualities about IINA; the third row suggests a different app than IINA.

**Comment by svobs** [comment]
Kind of odd to have different colors for the logo between large & small sizes. Might it not be more consistent to have separate colors for the larger icons as well? The large icons could still include the banners, but the smaller icons would drop them.
